nixos/modules/programs/xonsh.nix

```diff
@@ -32,6 +32,38 @@ in
         type = lib.types.lines;
       };
 
+      xontribs = lib.mkOption {
+        default = [ ];
+        type = lib.types.listOf lib.types.package;
+        description = ''
+          Add the listed xontribs to the package options. Available xontribs are
+          under xonsh.xontribs.
+
+          Take care in using this option along with manually defining the package
+          option above, as the two can result in conflicting sets of build dependencies.
+          This option assumes that the package option has an overridable argument
+          called `extraPackages`, so if you override the package option but also
+          intend to use this option, be sure that your resulting package still honors
+          the necessary option.
+        '';
+        example = "xontribs = with pkgs.xonsh.xontribs; [ xontrib-vox xontrib-abbrevs ];";
+      };
+
+      extraPackages = lib.mkOption {
+        default = (ps: [ ]);
+        type = lib.types.functionTo (lib.types.listOf lib.types.package);
+        description = ''
+          Add the specified extra packages to the xonsh package.
+          Preferred over using `programs.xonsh.package` as it composes with `programs.xonsh.xontribs`.
+
+          Take care in using this option along with manually defining the package
+          option above, as the two can result in conflicting sets of build dependencies.
+          This option assumes that the package option has an overridable argument
+          called `extraPackages`, so if you override the package option but also
+          intend to use this option, be sure that your resulting package still honors
+          the necessary option.
+        '';
+      };
```

Comment:
This was addressed in the review of my original PR. Xontribs are intentionally hobbled code and do not need to exclusively be written in Python (they can also be written in Xonsh syntax itself). Rather than using the established methods for Python plugin detection, Xonsh requires that its plugins all drop their entry point into the same folder and leave out the `__init__.py` file. Xonsh itself provides that init file and then looks for the xontribs in that directory. As such, xontribs are not truly valid Python packages and really shouldn't be handled as such.

Reply:
> This was addressed in the review of my original PR. Xontribs are intentionally hobbled code and do not need to exclusively be written in Python (they can also be written in Xonsh syntax itself). Rather than using the established methods for Python plugin detection, Xonsh requires that its plugins all drop their entry point into the same folder and leave out the `__init__.py` file. Xonsh itself provides that init file and then looks for the xontribs in that directory. As such, xontribs are not truly valid Python packages and really shouldn't be handled as such.

Xontribs and Python packages are both added to the wrapper's parameter `extraPackages` now and works fine so far for me, so I don't understand why there are two options to override `extraPackages`. Do you mean the wrapper needs rework?